### PR TITLE
#101

### LINKED ISSUE #101: Convert KgNodeSearchResultsPage to use mui-datatables

### PR DESCRIPTION
**Pull request by 123joshuawu**
Use `MUIDataTable` for `KgNodeSearchResultsPage`

Closes #101

### CODE REVIEW

File: gui/src/ts/components/kg/node/KgNodeTable.tsx [hunk 1]
```diff
@@ -25,66 +15,158 @@ const showListAsColumn = (list: string[]) =>
...
+  {
+    name: "other",
+    label: "Other",
+  },
+  {
+    name: "pos",
+    label: "Pos",
+    options: {
+      customBodyRender(pos) {
+        return pos ? showListAsColumn((pos as string).split(",")) : null;
```
Comment (gordom6) [inline gui/src/ts/components/kg/node/KgNodeTable.tsx:92]: This split and the datasources above will go away, these need to be parsed into lists on the server side. No need to change your code.

File: gui/src/ts/components/kg/node/KgNodeTable.tsx [hunk 2]
```diff
@@ -25,66 +15,158 @@ const showListAsColumn = (list: string[]) =>
...
+  page,
+  onChangePage,
+  onChangeRowsPerPage,
+}) => {
+  // https://github.com/gregnb/mui-datatables/issues/756
+  // Since the MUIDataTable has its own state, it ignores passed in values
+  // for page and rowsPerPage
+  // Workaround by checking the table state via ref and updating
+  // with internal api
+  const tableCallbackRef = (table: any) => {
```
Comment (gordom6) [inline gui/src/ts/components/kg/node/KgNodeTable.tsx:128]: You need this in order to keep the query string updated?

File: gui/src/ts/components/kg/node/KgNodeTable.tsx [hunk 3]
```diff
@@ -25,66 +15,158 @@ const showListAsColumn = (list: string[]) =>
...
+          <Typography variant="h6" data-cy="title">
+            {title}
+          </Typography>
+        }
+        data={nodes}
+        columns={columns}
+        options={{
+          count,
+          serverSide: true,
+          sort: false,
```
Comment (gordom6) [inline gui/src/ts/components/kg/node/KgNodeTable.tsx:161]: We'll need to implement sorting on the server side. I filed a new issue for that.

File: test/integration/cypress/integration/kg/KgNodeSearchResultsPage.spec.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  });
+
+  it("Should show rows per page", () => {
+    page.resultsTable.rowsPerPage.should("have.text", 10);
+  });
+
+  it("Pagination should not break", () => {
+    page.resultsTable.paginateNext();
+
+    page.resultsTable;
```
Comment (gordom6) [inline test/integration/cypress/integration/kg/KgNodeSearchResultsPage.spec.ts:55]: Should probably check that the nodes have changed here, but I'll let it slide.

File: gui/src/ts/components/kg/node/KgNodeTable.tsx (hunk 1 above)
Comment (123joshuawu) [reply gui/src/ts/components/kg/node/KgNodeTable.tsx:92]: Gotcha, will keep an eye out for that

File: gui/src/ts/components/kg/node/KgNodeTable.tsx (hunk 2 above)
Comment (123joshuawu) [reply gui/src/ts/components/kg/node/KgNodeTable.tsx:128]: This is used to update the table state if the query string updates (e.g. pressing back button or directly going to url).

File: gui/src/ts/components/kg/node/KgNodeTable.tsx (hunk 3 above)
Comment (123joshuawu) [reply gui/src/ts/components/kg/node/KgNodeTable.tsx:161]: Sounds good

**Comment by 123joshuawu** [comment]
@gordom6 Based on what I've read, it seems that currently under the hood React wraps all event handlers with `unstable_batchedupdates`, so the "unstable" is a bit misleading.

**Pull request merged**